common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -409,19 +472,34 @@ protected boolean runAllTasks(long timeoutNanos) {
             if ((runTasks & 0x3F) == 0) {
                 lastExecutionTime = ScheduledFutureTask.nanoTime();
                 if (lastExecutionTime >= deadline) {
-                    break;
+                    return DeadlineRunResult.TIMED_OUT;
```

Comment:
should we check here if we have run all tasks? if we have run all tasks claiming that we `TIMED_OUT` seems misleading.